### PR TITLE
447 components create icon component

### PR DESCRIPTION
**Pull request by andresz1**
## Add icon component

**TASK**: #447 

### Description, Motivation and Context
This PR:
- Add icon component
- Add `generate` npm command to create a new component

### Types of changes
- [ ] 🛠️ Tool
- [ ] 🪲 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [ ] 📷 Demo
- [ ] 🧪 Test
- [ ] 🧠 Refactor
- [ ] 💄 Styles

### Screenshots - Animations

https://user-images.githubusercontent.com/6877967/228169142-06348af8-f61a-4a5f-a9a8-3868b1ad5d84.mp4

### CODE REVIEW

File: packages/components/icon/package.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "access": "public"
+  },
+  "main": "./dist/index.js",
+  "module": "./dist/index.mjs",
+  "types": "./dist/index.d.ts",
+  "scripts": {
+    "build": "vite build"
+  },
+  "dependencies": {
+    "@radix-ui/react-visually-hidden": "1.0.2"
```
Comment (andresz1) [inline packages/components/icon/package.json:15]: This should be replace by our `VisuallyHidden` component when is done

Comment (andresz1) [reply packages/components/icon/package.json:15]: See #449

File: packages/components/icon/src/Icon.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  return (
+    <>
+      {React.cloneElement(child as React.ReactElement, {
+        className: iconVariants({ className, size }),
+        'aria-hidden': 'true',
+        focusable: 'false',
+        ...others,
+      })}
+
+      {label && <VisuallyHidden.Root>{label}</VisuallyHidden.Root>}
```
Comment (andresin87) [inline packages/components/icon/src/Icon.tsx:29]: Not needed. Icons are prepared to have a title in it. 

https://developer.mozilla.org/en-US/docs/Web/SVG/Element/title

File: packages/components/icon/src/Icon.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import * as VisuallyHidden from '@radix-ui/react-visually-hidden'
+import React, { PropsWithChildren } from 'react'
+
+import { iconVariants, IconVariantsProps } from './Icon.variants'
+
+export interface IconProps
```
Comment (acd02) [inline packages/components/icon/src/Icon.tsx:6]: Maybe we should ensure (with TS) that only one child can be passed to this component, like this:
```tsx
export interface IconProps extends IconVariantsProps, React.SVGProps<SVGElement> {
  /**
   * The accessible label for the icon. This label will be visually hidden but announced to screen
   * reader users, similar to `alt` text for `img` tags.
   */
  label?: string
  children: React.ReactElement
}
```

usage:
```tsx
<Icon> // 
  <svg></svg>
  <svg></svg>
</Icon>
```
⬆️ , you'll get a TS error
```ts
This JSX tag's 'children' prop expects a single child of type 'ReactElement<any, string | JSXElementConstructor<any>>', but multiple children were provided.
```

File: packages/components/icon/src/Icon.tsx (hunk 2 above)
Comment (andresz1) [reply packages/components/icon/src/Icon.tsx:29]: https://www.scottohara.me/blog/2019/05/22/contextual-images-svgs-and-a11y.html

**Review by acd02 — APPROVED** [review]

**Review by andresin87 — APPROVED** [review]

**Pull request merged**